src/ULog.ts

```diff
@@ -220,37 +259,73 @@ export class ULog {
     return this.#dataTimeRange;
   }
 
-  async #createIndex(): Promise<void> {
+  async #createIndex(reader: ChunkedReader): Promise<void> {
+    if (!this.#header) {
+      throw new Error(`Cannot read before open`);
+    }
+
     const timeIndex: IndexEntry[] = [];
     const dataCounts = new Map<number, number>();
     let minTimestamp: bigint | undefined;
     let maxTimestamp = 0n;
     let logMessageCount = 0;
 
-    let i = 0;
-    let message: DataSectionMessage | undefined;
-    while ((message = await this.#readParsedMessage())) {
-      if (message.type === MessageType.Data) {
-        if (minTimestamp == undefined || message.value.timestamp < minTimestamp) {
-          minTimestamp = message.value.timestamp;
+    for (;;) {
+      const offset = reader.position();
+
+      const rawMessage = await readRawMessage(reader, this.#dataEnd);
+      if (!rawMessage) {
+        break;
+      }
+
+      if (rawMessage.type === MessageType.AddLogged) {
+        this.#handleSubscription(rawMessage);
```

Comment:
this is also done in `#readParsedMessage`. I guess we want to do this only here?

Reply:
Ah yea - we don't need to do it in #readParsedMessage if we do it here